codec-http2/src/test/java/io/netty/handler/codec/http2/ReadOnlyHttp2HeadersTest.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    @Test
+    public void testPseudoHeadersMustComeFirstWhenIteratingServer() {
+        Http2Headers headers = newServerHeaders();
+        verifyPseudoHeadersFirst(headers);
+    }
+
+    @Test
+    public void testPseudoHeadersMustComeFirstWhenIteratingClient() {
+        Http2Headers headers = newClientHeaders();
+        verifyPseudoHeadersFirst(headers);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testIteratorReadOnlyClient() {
+        testIteratorReadOnly(newClientHeaders());
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testIteratorReadOnlyServer() {
+        testIteratorReadOnly(newServerHeaders());
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testIteratorReadOnlyTrailers() {
+        testIteratorReadOnly(newTrailers());
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testIteratorEntryReadOnlyClient() {
+        testIteratorEntryReadOnly(newClientHeaders());
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testIteratorEntryReadOnlyServer() {
+        testIteratorEntryReadOnly(newServerHeaders());
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testIteratorEntryReadOnlyTrailers() {
+        testIteratorEntryReadOnly(newTrailers());
+    }
+
+    @Test
+    public void testSize() {
+        Http2Headers headers = newTrailers();
+        assertEquals(otherHeaders().length / 2, headers.size());
+    }
+
+    @Test
+    public void testIsNotEmpty() {
+        Http2Headers headers = newTrailers();
+        assertFalse(headers.isEmpty());
+    }
+
+    @Test
+    public void testIsEmpty() {
+        Http2Headers headers = ReadOnlyHttp2Headers.trailers(false);
+        assertTrue(headers.isEmpty());
+    }
+
+    @Test
+    public void testContainsName() {
+        Http2Headers headers = newClientHeaders();
+        assertTrue(headers.contains("Name1"));
+        assertTrue(headers.contains(Http2Headers.PseudoHeaderName.PATH.value()));
+        assertFalse(headers.contains(Http2Headers.PseudoHeaderName.STATUS.value()));
+        assertFalse(headers.contains("a missing header"));
+    }
+
+    @Test
+    public void testContainsNameAndValue() {
+        Http2Headers headers = newClientHeaders();
+        assertTrue(headers.contains("Name1", "Value1"));
+        assertTrue(headers.contains(Http2Headers.PseudoHeaderName.PATH.value(), "/foo"));
+        assertFalse(headers.contains(Http2Headers.PseudoHeaderName.STATUS.value(), "200"));
+        assertFalse(headers.contains("a missing header", "a missing value"));
+    }
+
+    @Test
+    public void testGet() {
+        Http2Headers headers = newClientHeaders();
+        assertTrue(AsciiString.contentEqualsIgnoreCase("value1", headers.get("Name1")));
+        assertTrue(AsciiString.contentEqualsIgnoreCase("/foo",
```

Comment:
I don't think there is a strict style guide for line wrapping (other than 120 chars). Here is our check style file: https://github.com/netty/netty-build/blob/master/src/main/resources/io/netty/checkstyle.xml. Feel free to submit a PR if you have an idea for an improvement.